spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -473,5 +498,34 @@ object StandardizationJob {
     )
   }
 
+  /**
+    * Ensures that the 'spark.sql.columnNameOfCorruptRecord' Spark setting is set to field name not present in the
+    * output schema
+    * @param desiredSchema the desired output schema
+    * @param cmd           program settings
+    * @param spark         current Spark session
+    * @return              if columnNameOfCorruptRecord is to be used then the name of the column, if not and rather
+    *                      fail fast, function returns None
+    */
+  private def ensureUniqueCorruptRecordFieldName(desiredSchema: StructType, cmd: StdCmdConfig)
```

Comment:
Name: ideas welcome 😃 
Function: it might be doing 1.5 things 😉 It makes the special column name unique and returns that name. The 0.5 logic is that it actually returns `None` if the special column is not to be used. And my feeling is separating this logic outside would not help.

Reply:
`SchemaUtils.getClosestUniqueName()` does the unique field finding bit. But it uses a different suffix `_$i` instead of `$i`).